From: Pablo Neira Ayuso <pablo@netfilter.org>
Subject: py: remove setup.py integration with autotools

With Python distutils and setuptools going deprecated, remove
integration with autotools. This integration is causing issues
in modern environments.

Note that setup.py is still left in place under the py/ folder.

Signed-off-by: Pablo Neira Ayuso <pablo@netfilter.org>
Origin: upstream, https://marc.info/?l=netfilter-devel&m=168968163015356
---

diff --git a/Makefile.am b/Makefile.am
index 72fb4e88012d..84c3c366b86a 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -4,10 +4,8 @@ SUBDIRS = 	src	\
 		include	\
 		files	\
 		doc	\
-		examples
-if HAVE_PYTHON
-SUBDIRS += py
-endif
+		examples\
+		py
 
 EXTRA_DIST =	tests	\
 		files
diff --git a/configure.ac b/configure.ac
index adb782667438..b0201ac3528e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -108,25 +108,6 @@ AC_DEFINE([HAVE_LIBJANSSON], [1], [Define if you have libjansson])
 ])
 AM_CONDITIONAL([BUILD_JSON], [test "x$with_json" != xno])
 
-AC_ARG_ENABLE(python,
-       AS_HELP_STRING([--enable-python], [Enable python]),,[enable_python=check]
-       )
-
-AC_ARG_WITH([python_bin],
-            [AS_HELP_STRING([--with-python-bin], [Specify Python binary to use])],
-	    [PYTHON_BIN="$withval"], [AC_PATH_PROGS(PYTHON_BIN, python python2 python2.7 python3)]
-	   )
-
-AS_IF([test "x$PYTHON_BIN" = "x"], [
-	AS_IF([test "x$enable_python" = "xyes"], [AC_MSG_ERROR([Python asked but not found])],
-	[test "x$enable_python" = "xcheck"], [
-		AC_MSG_WARN([Python not found, continuing anyway])
-		enable_python=no
-	])
-])
-
-AM_CONDITIONAL([HAVE_PYTHON], [test "$enable_python" != "no"])
-
 AC_CONFIG_FILES([					\
 		Makefile				\
 		libnftables.pc				\
@@ -157,10 +138,3 @@ nft configuration:
   enable man page:              ${enable_man_doc}
   libxtables support:		${with_xtables}
   json output support:          ${with_json}"
-
-AS_IF([test "$enable_python" != "no"], [
-	echo "  enable Python:		yes (with $PYTHON_BIN)"
-	], [
-	echo "  enable Python:		no"
-	]
-	)
diff --git a/py/Makefile.am b/py/Makefile.am
index 215ecd9e4751..f10ae360599f 100644
--- a/py/Makefile.am
+++ b/py/Makefile.am
@@ -1,28 +1 @@
 EXTRA_DIST = setup.py __init__.py nftables.py schema.json
-
-all-local:
-	cd $(srcdir) && \
-		$(PYTHON_BIN) setup.py build --build-base $(abs_builddir)
-
-install-exec-local:
-	cd $(srcdir) && \
-		$(PYTHON_BIN) setup.py build --build-base $(abs_builddir) \
-		install --prefix $(DESTDIR)$(prefix)
-
-uninstall-local:
-	rm -rf $(DESTDIR)$(prefix)/lib*/python*/site-packages/nftables
-	rm -rf $(DESTDIR)$(prefix)/lib*/python*/dist-packages/nftables
-	rm -rf $(DESTDIR)$(prefix)/lib*/python*/site-packages/nftables-[0-9]*.egg-info
-	rm -rf $(DESTDIR)$(prefix)/lib*/python*/dist-packages/nftables-[0-9]*.egg-info
-	rm -rf $(DESTDIR)$(prefix)/lib*/python*/site-packages/nftables-[0-9]*.egg
-	rm -rf $(DESTDIR)$(prefix)/lib*/python*/dist-packages/nftables-[0-9]*.egg
-
-clean-local:
-	cd $(srcdir) && \
-		$(PYTHON_BIN) setup.py clean \
-		--build-base $(abs_builddir)
-	rm -rf scripts-* lib* build dist bdist.* nftables.egg-info
-	find . -name \*.pyc -delete
-
-distclean-local:
-	rm -f version
-- 
2.30.2

